### An Institution wishes to find out their student's ability in mathematics, reading and writing skills. The Institution wants to do an exploratory study to check the following information.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

### a. Find out how many males and females participated in the test.

In [2]:
df = pd.read_csv('Students_Performance.csv')

In [3]:
df['gender'].value_counts()['male']

482

In [4]:
df['gender'].value_counts()['female']

518

There are 518 females and 482 males participated in the test.

### b. What do you think about the students 'parental level of education'?

In [5]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

A larger percent of parents have some college or associate's degree. Only 59 parents have attained a master's degree.

### c. Who scores the most on average for math, reading and writing based on Gender and Test preparation course?

In [6]:
for i in ['math score','reading score','writing score']:
    average = df.groupby('gender')[i].mean()
    print(average)

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64
gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64
gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64


In Math, on average male students scored more and in reading and writing female students scored more on the basis of Gender.

In [4]:
for i in ['math score','reading score','writing score']:
    average = df.groupby('gender')[i].mean()
    print(average)

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64
gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64
gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64


All the students who completed the course have scored better marks on an average than those who did not complete.

### d. What do you think about the scoring variation for math,reading and writing based on Gender and Test preparation course?

In [5]:
for i in ['math score','reading score','writing score']:
    variance = df.groupby('gender')[i].var()
    print(variance)

gender
female    256.958593
male      211.889097
Name: math score, dtype: float64
gender
female    207.677438
male      200.211010
Name: reading score, dtype: float64
gender
female    220.369327
male      202.413924
Name: writing score, dtype: float64


In [6]:
for i in ['math score','reading score','writing score']:
    variance = df.groupby('test preparation course')[i].var()
    print(variance)

test preparation course
completed    210.884027
none         246.668662
Name: math score, dtype: float64
test preparation course
completed    183.265864
none         213.419851
Name: reading score, dtype: float64
test preparation course
completed    175.202612
none         226.251739
Name: writing score, dtype: float64


### e. The top 25% of students on their math score.

In [8]:
df.sort_values(by=['math score'],ascending=False).head(int(len(df['math score'])*(25/100)))

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]

## case study on Testing of Hypothesis

The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

In [5]:
import pandas as pd
import numpy as np
import scipy.stats as stats
from statsmodels.stats.weightstats import ztest

data = pd.read_csv('Sales_add.csv')
data


Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [4]:
alpha = 0.05
t_val,p_val = stats.ttest_rel(data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'])

print(p_val)



6.336667004575778e-11


In [5]:
if p_value < alpha:
    print('Reject null hypothesis: The digital marketing strategy has increased sales')
else:
    print('Do not reject null hypothesis: The digital marketing strategy does not have any effect on sales')

Reject null hypothesis: The digital marketing strategy has increased sales


The digital marketing strategy has increased sales.

### Whether there is any dependency between the features Region and Manager.

In [6]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
data = pd.read_csv('Sales_add.csv')

In [7]:
data= data.rename(columns={'Sales_After_digital_add(in $)':'Sales'})
data

Month      Region      Manager  Sales_before_digital_add(in $)   Sales
0    Month-1  Region - A  Manager - A                          132921  270390
1    Month-2  Region - A  Manager - C                          149559  223334
2    Month-3  Region - B  Manager - A                          146278  244243
3    Month-4  Region - B  Manager - B                          152167  231808
4    Month-5  Region - C  Manager - B                          159525  258402
5    Month-6  Region - A  Manager - B                          137163  256948
6    Month-7  Region - C  Manager - C                          130625  222106
7    Month-8  Region - A  Manager - A                          131140  230637
8    Month-9  Region - B  Manager - C                          171259  226261
9   Month-10  Region - C  Manager - B                          141956  193735
10  Month-11  Region - B  Manager - C                          159339  203175
11  Month-12  Region - A  Manager - A                          178939  276279
12  Month-13  Region - B  Manager - A                          145062  205180
13  Month-14  Region - A  Manager - A                          151514  253131
14  Month-15  Region - C  Manager - A                          147463  229336
15  Month-16  Region - A  Manager - B                          177195  187305
16  Month-17  Region - A  Manager - C                          140862  234908
17  Month-18  Region - C  Manager - B                          167996  191517
18  Month-19  Region - B  Manager - A                          132135  227040
19  Month-20  Region - A  Manager - B                          152493  212579
20  Month-21  Region - B  Manager - A                          147425  263388
21  Month-22  Region - A  Manager - C                          130263  243020

In [8]:
model = ols('Sales ~C(Region)+C(Manager) + C(Region) : C(Manager)', data=data).fit()
anova_table = sm.stats.anova_lm(model, type = 2)
print(anova_table)

                        df        sum_sq       mean_sq         F    PR(>F)
C(Region)              2.0  1.370211e+09  6.851056e+08  0.996435  0.395713
C(Manager)             2.0  2.568861e+09  1.284431e+09  1.868109  0.193579
C(Region):C(Manager)   4.0  8.388157e+08  2.097039e+08  0.304999  0.869479
Residual              13.0  8.938238e+09  6.875567e+08       NaN       NaN


In [10]:
alpha = 0.05
if any (anova_table['PR(>F)']<alpha):
    print('There is significant interaction between factors')
    
    Region = anova_table.loc['C(Region)','PR(>F)']<alpha
    Manager = anova_table.loc['C(Manager)','PR(>F)']<alpha
    
    if Region:
        print('Region has significant effect')
    else:
        print('Region has no significant effect')
    if Manager:
        print('Manager has significant effect')
    else:
        print('Manager has no significant effect')
else:
         print('No significant interaction between factors')

No significant interaction between factors


There is no dependency between Region and Manager.